Library/Homebrew/language/node.rb

```diff
@@ -22,8 +22,19 @@ def self.std_npm_install_args(libexec)
       # tell npm to not install .brew_home by adding it to the .npmignore file
       # (or creating a new one if no .npmignore file already exists)
       open(".npmignore", "a") { |f| f.write("\n.brew_home\n") }
+      # npm 5.0.0 broke the assumption Homebrew made that npm would always
+      # create and install real directories when passed a file path. We can
+      # restore the old behaviour by packing the path first with "npm pack"
+      # which also helpfully spits out the filename it just created allowing
+      # us to pass that as the final argument.
+
       # npm install args for global style module format installed into libexec
-      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+      %W[
+        --verbose
+        --global
+        --prefix=#{libexec}
+        #{Dir.pwd}/#{Utils.popen_read("npm pack").chomp}
```

Comment:
Are you sure this was possible? Would `std_npm_install_args(libexec/"vendor") << "foo@2"` really overide the `"."`paramater which tells npm to install the local folder (or at least told before npm@5)?

I think it would just tell npm to install to local folder = `"."` and `foo@2`and if couldn't find a valid node module in the current working directory it would error out early.
I don't think that such a hacky behavior (installing the module at the current local folder and an other named module) is something we want to support here. Also it looks to contradict against the homebrew philosopy that foo should be vendored as a resource too in such a use case.